source/cheby.f90

```diff
@@ -1,41 +1,55 @@
-  real(kind=fPrec), save :: cheby_offset_x(ncheby), cheby_offset_y(ncheby)  ! hor./vert. offset [mm]
-  real(kind=fPrec), save :: cheby_angle(ncheby)       ! rotation angle about the longitudinal axis [rad]
-  integer, save          :: cheby_itable(ncheby)      ! index of chebyshev table
-  real(kind=fPrec), save :: cheby_scalingFact(ncheby) ! scaling factor []
+  integer,          allocatable, save :: cheby_itable(:)      ! index of chebyshev table
+  real(kind=fPrec), allocatable, save :: cheby_r2(:)          ! outer radius R2 [mm] (optional)
+  real(kind=fPrec), allocatable, save :: cheby_r1(:)          ! inner radius R1 [mm] (optional)
+  real(kind=fPrec), allocatable, save :: cheby_angle(:)       ! rotation angle about the longitudinal axis [deg] (optional)
+  real(kind=fPrec), allocatable, save :: cheby_offset_x(:)    ! hor. offset [mm] (optional)
+  real(kind=fPrec), allocatable, save :: cheby_offset_y(:)    ! ver. offset [mm] (optional)
+  real(kind=fPrec), allocatable, save :: cheby_I(:)           ! actual powering of lens [A] (optional)
+  real(kind=fPrec), allocatable, save :: cheby_scalingFact(:) ! scaling factor [] (computed internally)
+  
+  ! show map
+  integer,          allocatable, save :: cheby_iLens(:)       ! lens for which a map echo is requested
+  character(len=:), allocatable, save :: cheby_mapFileName(:) ! file name
+  real(kind=fPrec), allocatable, save :: cheby_mapXmin(:)     ! xMin [mm]
+  real(kind=fPrec), allocatable, save :: cheby_mapXmax(:)     ! xMax [mm]
+  integer,          allocatable, save :: cheby_mapNx(:)       ! number of intervals
+  real(kind=fPrec), allocatable, save :: cheby_mapYmin(:)     ! yMin [mm]
+  real(kind=fPrec), allocatable, save :: cheby_mapYmax(:)     ! yMax [mm]
+  integer,          allocatable, save :: cheby_mapNy(:)       ! number of intervals
 
   ! tables with chebyshev coefficients
-  integer, parameter     :: cheby_unit=-1             ! unit for reading the chebyshev coefficients
-  integer, parameter     :: cheby_max_order=60        ! max order of chebyshev polynomials
-  integer, parameter     :: cheby_lFileName=16        ! length of filenames
-  character(len=cheby_lFileName), save:: cheby_filename(ncheby_tables)  ! file names
-  real(kind=fPrec), save :: cheby_coeffs(0:cheby_max_order,0:cheby_max_order,ncheby_tables) ! coefficients
-  integer, save          :: cheby_maxOrder(ncheby_tables)  ! max order of the current map
-  real(kind=fPrec), save :: cheby_refRadius(ncheby_tables) ! reference radius [mm]
-
+  character(len=:), allocatable, save :: cheby_filename(:)    ! file names
+  real(kind=fPrec), allocatable, save :: cheby_coeffs(:,:,:)  ! coefficients
+  integer,          allocatable, save :: cheby_maxOrder(:)    ! max order of the current map
+  real(kind=fPrec), allocatable, save :: cheby_refI(:)        ! reference current [A] (optional)
+  real(kind=fPrec), allocatable, save :: cheby_refR(:)        ! reference radius [mm] (mandatory)
```

Comment:
All of these should state variable declarations should have public or private keyword as well. Private should be the default, and public only used if the variables have to be accesses outside of the module.